python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -419,10 +418,98 @@ def deduplicates(self):
             data_shards = spark_df_to_pd_sparkxshards(distinctDF)
             return data_shards
         else:
-            # we may support numpy or other types later
             invalidInputError(False,
                               "Currently only support dedup() on XShards of Pandas DataFrame")
 
+    def sort_values(self, col_names=None, ascending=True):
+        """
+        Sort the value of shards. This is only applicable for SparkXShards of Pandas Series.
+        
+        :param col_names list of column or column naames to sort by
+        :param ascending bool, default True. Specify sort orders
+        :return: a new SparkXShards sorted by the specified columns.
+        """
+        if self._get_class_name() == 'pandas.core.frame.DataFrame':
+            import pandas as pd
+            df = self.to_spark_df()
+            sort_df = df.sort(col_names, ascending=ascending)
+            sort_df = sort_df.coalesce(1)
+            data_shards = spark_df_to_pd_sparkxshards(sort_df)
+            return data_shards
+        else:
+            invalidInputError(False,
+                              "Currently only support sort() on XShards of Pandas DataFrame")
+
+    def max_values(self, col_name):
+        """
+        Get the max values of the column name. This is only applicable for SparkXShards
+         of Pandas Series.
+
+        :param col_name column name that need return the max value
+        :return: maximum for the specified columns.
+        """
+        if self._get_class_name() == 'pandas.core.frame.DataFrame':
+            import pandas as pd
+            df = self.to_spark_df()
+            from pyspark.sql.functions import max
+            max_value = df.select(max(col_name)).collect()
+            return max_value
+        else:
+            invalidInputError(False,
+                              "Currently only support max() on XShards of Pandas DataFrame")
+
+    def get_null_sum(self):
+        """
+        With SparkXShards of pandas data frame, the api will get null numbers for
+        each column. For other type of SparkXShards, it will throw exception
+
+       :return: SparkXShards of pandas data frame with 2 columns, `col` represents column name,
+        `total` represents null numbers
+        """
+        if self._get_class_name() != 'pandas.core.frame.DataFrame':
+            invalidInputError(False,
+                              "Currently only support assembleFeatureLabelCols() on"
+                              " XShards of Pandas DataFrame")
+
+        def get_na_sum(df):
+            import pandas as pd
+            series = df.isnull().sum()
+            df = pd.DataFrame({'col': series.index, 'total': series.values})
+            return df
+
+        na_cnt_shard = self.transform_shard(get_na_sum)
```

Comment:
use `rdd.map_partition` instead of `transform_shard` to reduce unnecessary caching

Reply:
Fixed.